src/lib.rs

```diff
@@ -2137,6 +2137,20 @@ pub trait ConfigIterParser<'a, I: Input<'a>, O, E: ParserExtra<'a, I> = extra::D
             phantom: PhantomData,
         }
     }
+
+    /// A combinator that allows fallible configuration of the parser from the current context -
+    /// if an error is returned, parsing fails.
+    fn try_configure<F>(self, cfg: F) -> TryIterConfigure<Self, F, O>
```

Comment:
Could this behaviour be replicated with `try_map` on the output of the parser that provides configuration? Or is this for a sub-branch of the context-consuming parser?

Reply:
You can replicate it that way, yeah. It has a bit of the filter then map issue though - you validate the input, then need to either have an `unreachable` in the configuration, or map it to another type which can't have the relevant states.